plugin/pkg/admission/priority/admission.go

```diff
@@ -196,8 +196,14 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 			pod.Spec.PriorityClassName = pcName
 		} else {
 			pcName := pod.Spec.PriorityClassName
-			if !priorityClassPermittedInNamespace(pcName, a.GetNamespace()) {
-				return admission.NewForbidden(a, fmt.Errorf("pods with %v priorityClass is not permitted in %v namespace", pcName, a.GetNamespace()))
+			// If ResourceQuotaScopeSelectors is enabled, we should let critical priorityClass to be created
+			// any namespace where administrator wants it to be created.
+			// TODO: @ravig Add a default quota at cluster bootstrap which limits critical priorityClasses
```

Comment:
I think we would desire that quota is required mandatorily to be present in a namespace for allowing `SystemPriorityClasses` to be created in that namespace. To enforce this, `AdmissionConfiguration` object with matching scopes for `SystemPriorityClasses` must be configured.

Then to allow creation, quota can be created in desired namespaces.

Reply:
The problem with this approach is that users will need to add extra configuration to their clusters when they enable ResourceQuotaScopeSelectors. If they don't, this change may create a security gap in their clusters.
Whatever solution we propose here should not need extra configuration by users.